File: app/src/main/java/com/stan/dagger_android/base/BaseActivity.java
package com.stan.dagger_android.base;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;

import butterknife.ButterKnife;
import dagger.android.AndroidInjection;
import dagger.android.support.DaggerAppCompatActivity;

/**
 * Created by uu on 2017/12/28.
 */

public class BaseActivity extends DaggerAppCompatActivity{
    @Override
    public void setContentView(View view) {
        super.setContentView(view);
        ButterKnife.bind(this);
    }


    @Override
    public void setContentView(int layoutResID) {
        super.setContentView(layoutResID);
        ButterKnife.bind(this);
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        AndroidInjection.inject(this);
        super.onCreate(savedInstanceState);
    }
}

File: app/src/main/java/com/stan/dagger_android/base/BaseActivity.kt
package com.stan.dagger_android.base

import android.view.View

import butterknife.ButterKnife
import dagger.android.support.DaggerAppCompatActivity

/**
 * Created by uu on 2017/12/28.
 */

open class BaseActivity : DaggerAppCompatActivity() {
    override fun setContentView(view: View) {
        super.setContentView(view)
        ButterKnife.bind(this)
    }


    override fun setContentView(layoutResID: Int) {
        super.setContentView(layoutResID)
        ButterKnife.bind(this)
    }
}
